### PR TITLE
By default, ``make latexpdf`` will try to ignore LaTeX errors

### PR DESCRIPTION
**Pull request by jfbu**
Let ``latexmk`` work by default with `-f` option which forces it to try to compile to PDF, even in case of LaTeX errors. The call to ``make latexpdf`` or ``make -C <builddir>/latex`` can override this if ending with ``LATEXOPTS = "<options>"``.

Also, the ``all`` target of the LaTeX Makefile put in build directory has been moved higher up so that it is default target if ``make -C <builddir>/latex``  does not specify a target (formerly ``all-dvi`` was default, but for ``xelatex`` engine this resulted in no compilation attempted).

### Feature or Bugfix

This is a potential bugfix, as the feature is part of 1.6-release and not yet released.

I checked the project mentioned at #3289 (https://developer.blender.org/project/view/53/), and realized behaviour with 1.6-release differed from 1.5.5, due to the ``--halt-on-error`` from #3082 (409605d). With this PR, the ``make latexpdf`` behaves more like in ``1.5.5``.

The idea of the ``--halt-on-error`` was to facilitate people not too much knowledgeable in LaTeX identify the first error occurring in document, because secondary errors are sometimes misleading. But the problem is that LaTeX has sometimes minor issues causing errors, and it might be important for a project to still be able to produce a PDF file, even though LaTeX complains. Hence, this PR takes the opposite view to use by default a "force" mode.

This is not related to the ``sphinx-build`` processing to ``latex`` target but only regards the ``latex --> pdf`` part. 

Also, the Blender project uses a Makefile which after having done ``sphinx-build`` does``make -C <builddir>/latex`` and this caused problem in my testing with 1.6-release and `xelatex` engine due to ``make`` then attempting to use ``all-dvi`` which was not suitable. Hence this PR moves the ``all`` target (which defaults to ``all-pdf``) to be first in Makefile put in ``<builddir>/latex``.

### Relates
- #3082

### CODE REVIEW

**Review by tk0miya — APPROVED** [review]
+1

**Comment by jfbu** [comment]
Thanks @tk0miya for review 

I need your further opinion about whether I should put the ``-f`` into a specialized Make variable ``LATEXMKOPTS`` rather than ``LATEXOPTS``. Indeed for example the Blender Makefile does ``LATEXOPTS="-interaction nonstopmode"``, thus this overrides the ``-f`` in this PR.

- In the case of `xelatex` engine, this means `xdv` file (for which no viewer exists) is produced, but the next latexmk step of producing `pdf` file is not attempted if an error was encountered in latex processing, because the ``-f`` was lost due to ``LATEXOPTS`` being redefined,
- in general case, the processing will stop after first run of `pdflatex` hence table of contents, index, ..., will not be finalized, because latex error was encountered and latexmk was not run with `-f` option.

For these reasons, I think I should put the``-f`` into its own ``LATEXMKOPTS``.

The secondary issue is that it seems `latexmk` accepts `-interaction=nonstopmode` option but not the `-interaction nonstopmode` pdflatex option (with a space) as above, even if quoted, so scripts done by people for projects with ``LATEXOPTS`` configuration as in Blender case will not work out of the box if ``LATEXOPTS`` is fed to ``latexmk`` as is. I have pinged latexmk's author about advice on this point.

It seems for backwards compatibility I should make it so that ``LATEXOPTS`` contents get into the ``pdflatex`` command as is set-up in the `latexmkrc` file. The `%O` placeholder is already there for Latexmk's view of what options to use for `pdflatex` depending on what options it (latexmk) received, e.g. the `-silent` option is converted in the various appropriated `%O` for the executables latexmk calls. I think I should have ``LATEXOPTS`` inserted before the `%O`, then.

In short, for backwards compatibility the ``LATEXOPTS`` should not be option of ``latexmk`` because there is no 100% matching of option, but option of the pdflatex, etc... calls.

**Comment by tk0miya** [comment]
Did you mean the following? It's reasonable!
```
LATEXMKOPTS = -f
LATEXOPTS = --interaction=nonstopmode
$(PDFLATEX) $(LATEXMKOPTS) $(LATEXOPTS) '$<'
```

**Comment by jfbu** [comment]
Yes, but it is more complicated because in fact `$(PDFLATEX)` stands for `latexmk` hence `$(LATEXOPTS)` must end up as option of the real `pdflatex` call which is configured in `latexmkrc`. I will update.

**Comment by tk0miya** [comment]
okay, go ahead!

**Review by tk0miya — APPROVED** [review]
+1

**Comment by jfbu** [comment]
Thanks for feedback @tk0miya ! I have now separated ``latexmk`` and ``pdflatex`` options and it works fine in my testing. See the doc at https://github.com/sphinx-doc/sphinx/pull/3695/commits/ca543eee23dcc5593cea074fe069a8fbf8844eac#diff-9b024f3a65a9155bebddd27a7ed6de28R205. With this the Blender Makefile which does 
```
	$(SPHINXBUILD) -b latex ./manual "$(BUILDDIR)/latex"
	make -C "$(BUILDDIR)/latex" LATEXOPTS="-interaction nonstopmode"
```
works with no modification with 1.6.

Thanks @tk0miya! you reacted while I add forgotten to click "Comment" with this text, so I will now merge.
